feat(flows): Add equals and not_equals to the ffmpegCommandRemoveStreamByProperty plugin

Ev-Mu opened the pull request:
# ffmpegCommandRemoveStreamByProperty

This PR adds the `equals` and `not_equals` conditions to the [ffmpegCommandRemoveStreamByProperty](https://docs.tdarr.io/docs/plugins/flow-plugins/index/ffmpegCommand/Remove%20Stream%20By%20Property) plugin. This allows for more detailed control of what streams get removed. I copied the much more descriptive tooltips from the matching inputs of the [checkStreamProperty](https://docs.tdarr.io/docs/plugins/flow-plugins/index/file/Check%20Stream%20Property) plugin to help users understand what this plugin does.

## Reason why this change is important

I was attempting to remove all audio `codec_name`'s except `eac3` and in my list of other codes I want to get rid of, I have `ac3`. The `includes` check causes `ac3` to end up removing `eac3`. What I am trying to do is impossible with the current plugin as far as I can tell. I have added an image below to give more context of what I am trying to achieve.
<img width="611" height="527" alt="plugin_config" src="https://github.com/user-attachments/assets/15a4b0ca-75ff-4245-bd8c-9f26818a2c71" />

## Improvements beyond this

The [checkStreamProperty](https://docs.tdarr.io/docs/plugins/flow-plugins/index/file/Check%20Stream%20Property) plugin has an additional input for `Stream Type` (`all`, `video`, `audio`, `subtitle`, `data`). Adding this capability to this plugin would allow for even greater control and make the plugin even easier to user.

---

**Please let me know if you have any feedback or if there is a better way to achieve what I am trying to do. Thanks!**

Ev-Mu committed:
Convert to use exact match instead of substring

Ev-Mu committed:
Expanded the control of stream removal with more conditions

Fish2 commented:
I am interested in this as if you want to remove just ac3 it will remove both eac3 and ac3

HaveAGitGat committed:
Merge remote-tracking branch 'origin/master' into pr/880

HaveAGitGat committed:
chore: rebuild ffmpegCommandRemoveStreamByProperty index.js after merge

HaveAGitGat commented:
@Fish2 this PR adds `equals` which would handle that correctly. `includes` is unchanged and `eac3` would include `ac3` (which is the purpose of that option).

HaveAGitGat committed:
refactor: drop unreachable empty-input validation in ffmpegCommandRemoveStreamByProperty

loadDefaultValues rewrites '' back to each input's defaultValue ('codec_name'
and 'aac'), so the empty-string guards could never fire. Remove them rather
than carry dead code.

HaveAGitGat committed:
test: cover equals and not_equals conditions in ffmpegCommandRemoveStreamByProperty

Add 6 cases verifying:
- equals matches exact values only (ac3 does not remove eac3)
- equals supports multiple values via OR
- equals is case insensitive
- not_equals removes everything that is not the value
- not_equals over multiple values means equalling NONE of them
- not_equals is case insensitive

HaveAGitGat commented:
Added some improvements and tests. LGTM thanks
